src/hidra/receiver/asapo_transfer.py

```diff
@@ -33,7 +36,8 @@ def __init__(self, signal_host, target_host, detector_id,
         self.endpoint = endpoint
         self.beamtime = beamtime
         self.beamline = beamline
-        self.token = token
+        with open(token_path, "r") as f:
+            self.token = f.readline().split("\n")[0]
```

Comment:
Better do this outside of the constructor.

Reply:
Done.